### ISSUE TITLE
I'm here to ask you a question again.

### ISSUE THREAD

**Issue by Joon7330** [thestephencasper/feature_level_adv#3]
Hi, I am a student studying AI.

In custom_loss_patch_adv function and custom_loss_region_gen_patch_adv function.
There are some lam_ variables here.
I wonder what lam means in this code.
Can you explain this part in detail?

I'd appreciate it if you could answer me.

**Comment by thestephencasper** [comment]
All are coefficients for terms in the loss function for generating the adversarial features. Higher values mean more weight on those parts of the objective. Equation 2 in the paper is about this. 
lam_xent= weight on the cross entropy loss term. This measures how effectively the adversarial feature fools the model.
lam_tvar= weight on the total variation loss term. This measures how smooth/non-pixelated the feature is.
lam_disc= weight on the discriminator loss term. This measures how realistic the discriminator thinks the feature is.
lam_patch_xent= weight on the patch cross entropy loss term. This measures how effectively the adversarial feature disresembles the target glass. 
lam_ent= weight on the entropy loss term. This measures how effectively the adversarial feature successfully resembles something other than the target class.
